assets/dev/js/frontend/handlers/video.js

```diff
@@ -66,10 +71,79 @@ export default class Video extends elementorModules.frontend.handlers.Base {
 		this.getLightBox().setVideoAspectRatio( this.getElementSettings( 'aspect_ratio' ) );
 	}
 
+	prepareYTVideo( YT, onOverlayClick ) {
+		const elementSettings = this.getElementSettings(),
+			playerOptions = {
+				videoId: this.videoID,
+				events: {
+					onReady: () => {
+						if ( elementSettings.mute ) {
+							this.player.mute();
+						}
+
+						if ( elementSettings.autoplay || onOverlayClick ) {
+							this.player.playVideo();
+						}
+					},
+					onStateChange: ( event ) => {
+						if ( event.data === YT.PlayerState.ENDED && elementSettings.loop ) {
+							this.player.seekTo( elementSettings.start || 0 );
+						}
+					},
+				},
+				playerVars: {
+					controls: elementSettings.controls ? 1 : 0,
+					rel: elementSettings.rel ? 1 : 0,
+					playsinline: elementSettings.play_on_mobile ? 1 : 0,
+					modestbranding: elementSettings.modestbranding ? 1 : 0,
+					autoplay: elementSettings.autoplay ? 1 : 0,
+					start: elementSettings.start,
+					end: elementSettings.end,
+				},
+			};
+
+		if ( elementSettings.start ) {
+			playerOptions.start = elementSettings.start;
+		}
+
+		if ( elementSettings.end ) {
+			playerOptions.end = elementSettings.end;
+		}
+
+		if ( elementSettings.yt_privacy ) {
+			playerOptions.host = 'https://www.youtube-nocookie.com';
+			playerOptions.playerVars.origin = window.location.hostname;
+		}
+
+		this.player = new YT.Player( this.elements.$video[ 0 ], playerOptions );
+	}
+
 	bindEvents() {
 		this.elements.$imageOverlay.on( 'click', this.handleVideo.bind( this ) );
 	}
 
+	onInit() {
+		super.onInit();
+
+		const elementSettings = this.getElementSettings();
+
+		if ( 'youtube' !== elementSettings.video_type ) {
```

Comment:
It's recommended to create a VideoBase class that each provider can extend it.

Reply:
Will be considered as a maintenance task